crates/cxx-qt-lib-extras/src/core/qlocale.rs

```diff
@@ -0,0 +1,568 @@
+    }
+
+    #[namespace = "Qt"]
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qt.h");
+        type DayOfWeek = cxx_qt_lib::DayOfWeek;
+    }
+
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qstring.h");
+        type QString = cxx_qt_lib::QString;
+
+        include!("cxx-qt-lib/qstringlist.h");
+        type QStringList = cxx_qt_lib::QStringList;
+
+        include!("cxx-qt-lib-extras/qlocale.h");
+        type QLocale = super::QLocale;
+
+        /// Returns the localized name of the "AM" suffix for times specified using the conventions of the 12-hour clock.
+        #[rust_name = "am_text"]
+        fn amText(self: &QLocale) -> QString;
+
+        /// Returns the BCP47 field names joined with dashes.
+        // QString QLocale::bcp47Name(QLocale::TagSeparator separator = TagSeparator::Dash) const
+        #[rust_name = "bcp47_name"]
+        fn bcp47Name(self: &QLocale, separator: QLocaleTagSeparator) -> QString;
+
+        /// Returns the locale to use for collation.
+        #[rust_name = "collation"]
+        fn collation(self: &QLocale) -> QLocale;
+
+        #[rust_name = "create_separated_list"]
+        fn createSeparatedList(self: &QLocale, list: &QStringList) -> QString;
+
+        /// Returns a currency symbol according to the format.
+        // QString QLocale::currencySymbol(QLocale::CurrencySymbolFormat format = CurrencySymbol) const
+        #[rust_name = "currency_symbol"]
+        fn currencySymbol(self: &QLocale, format: QLocaleCurrencySymbolFormat) -> QString;
+
+        /// Returns the date format used for the current locale.
+        #[rust_name = "date_format"]
+        fn dateFormat(self: &QLocale, format: QLocaleFormatType) -> QString;
+
+        /// Returns the date time format used for the current locale.
+        #[rust_name = "date_time_format"]
+        fn dateTimeFormat(self: &QLocale, format: QLocaleFormatType) -> QString;
+
+        /// Returns the localized name of the day (where 1 represents Monday, 2 represents Tuesday and so on), in the format specified by type.
+        // QString QLocale::dayName(int day, QLocale::FormatType type = LongFormat) const
+        #[rust_name = "day_name"]
+        fn dayName(self: &QLocale, day: i32, formatType: QLocaleFormatType) -> QString;
+
+        /// Returns the fractional part separator for this locale.
+        #[rust_name = "decimal_point"]
+        fn decimalPoint(self: &QLocale) -> QString;
+
+        /// Returns the exponent separator for this locale.
+        #[rust_name = "exponential"]
+        fn exponential(self: &QLocale) -> QString;
+
+        /// Returns the first day of the week according to the current locale.
+        // Qt::DayOfWeek QLocale::firstDayOfWeek() const
+        #[rust_name = "first_day_of_week"]
+        fn firstDayOfWeek(self: &QLocale) -> DayOfWeek;
+
+        // QString QLocale::formattedDataSize(qint64 bytes, int precision = 2, QLocale::DataSizeFormats format = DataSizeIecFormat) const
```

Comment:
DataSizeFormats is QFlags, do we have prior art on how to model that?

Reply:
We unfortunately don't have anything for flags yet i don't think, we support enums but not flags.

@LeonMatthesKDAB something to figure out for 0.8 ?

Reply:
Right, sounds useful to find an ergonomic way to use Flags for 0.8. We don't have anything yet though.